docs/hooks/use-submit.md

```diff
@@ -78,6 +78,48 @@ formData.append("cheese", "gouda");
 submit(formData);
 ```
 
+### Payload Serialization
+
+You may also submit raw JSON to your `action` and the default behavior will be to encode the key/values into `FormData`:
+
+```tsx
+let obj = { key: "value" };
+submit(obj); // -> request.formData()
```

Comment:
Should there be additional type-checking for this use case (and I guess when explicitly setting `encType: 'application/x-www-form-urlencoded'`)?

If I understand correctly, it seems like a big footgun that I can now pass any type here and I won't get a type error and then it'll blow up later when the code assumes it can be serialized to FormData.

Reply:
Probably 😬 .  #10362 starts adding some more advanced type checks around `payload`/`encType`/`action` so we can try to add it there - but it's proven a bit tricky thus far to get the inferred types working right across params.  I'm sure we can figure something out though.  I'll drop a note over there